src/com.company/heap/MishaAndCandies.java

```diff
@@ -0,0 +1,115 @@
+package heap;
+
+import java.util.PriorityQueue;
+import java.util.Scanner;
+
+public class MishaAndCandies {
+
+    private int solve(int[] array, int b) {
+        int count = 0;
+        PriorityQueue<Integer> priorityQueue = new PriorityQueue<>();
+        for (int i = 0; i < array.length; i++) {
+            priorityQueue.add(array[i]);
+        }
+        for (int i = 0; i < priorityQueue.size(); i++) {
+            int minElement = priorityQueue.poll();
+            if (minElement <= b) {
+                int chocolateEatenCount = (int) Math.floor(minElement / 2);
```

Comment:
Math.floor is not required as division by 2 will always give floor value. Also typecasting to int is also not required which will be handled automatically.